src/Dialect/ONNX/ONNXOps/Canonicalize.cpp

```diff
@@ -1633,66 +1635,266 @@ struct RecomposeConcatPattern : public OpRewritePattern<ONNXConcatOp> {
+      if (!scaleOrBiasType || !scaleOrBiasType.hasStaticShape()) {
+        return rewriter.notifyMatchFailure(
+            matchOp, "bias/scale has not a static shape");
+      }
+
+      SmallVector<int64_t> biasOrScaleRankFixedShape;
+      biasOrScaleRankFixedShape.append(
+          addOrMulResultType.getRank() - scaleOrBiasType.getRank(), 1);
+      biasOrScaleRankFixedShape.append(
+          scaleOrBiasType.getShape().begin(), scaleOrBiasType.getShape().end());
+
+      // biasOrScaleRankFixedShape should have exactly one dimension that is not
+      // one
+      std::optional<int64_t> afterReshapeComputationDim;
+      for (auto [idx, dimSize] : enumerate(biasOrScaleRankFixedShape)) {
+        if (dimSize != 1) {
+          if (afterReshapeComputationDim) {
+            return rewriter.notifyMatchFailure(
+                matchOp, "scale/bias has more than one non-one dimension");
+          }
+          afterReshapeComputationDim = idx;
+        }
+      }
+      if (!afterReshapeComputationDim) {
+        return rewriter.notifyMatchFailure(
+            matchOp, "scale/bias has no non-one dimension");
+      }
+
+      const auto shapeIncludingComputationDim =
+          ArrayRef<int64_t>(reshapeResultType.getShape())
+              .slice(0, *afterReshapeComputationDim + 1);
+      const uint64_t computationRelevantSize = std::accumulate(
+          shapeIncludingComputationDim.begin(),
+          shapeIncludingComputationDim.end(), 1, std::multiplies<uint64_t>());
+
+      // The bias/scale dim should be not affected by the reshape. We need to
+      // map it back through it.
+      size_t reshapeInComputationDim;
+      auto reshapeInType =
+          dyn_cast<ShapedType>(reshapeOp->getOperand(0).getType());
+      if (!reshapeInType || !reshapeInType.hasStaticShape()) {
+        return rewriter.notifyMatchFailure(
+            matchOp, "reshape input has not a static shape");
+      }
+      const auto reshapeInShape = reshapeInType.getShape();
+
+      // trace the dim through the reshape
+      uint64_t acc = 1;
+      for (auto [idx, dimSize] : enumerate(reshapeInShape)) {
+        acc *= dimSize;
+        if (acc == computationRelevantSize) {
+          if (dimSize !=
+              biasOrScaleRankFixedShape[*afterReshapeComputationDim]) {
+            return rewriter.notifyMatchFailure(matchOp,
+                "bias/scale shape is not compatible with reshape input");
+          }
+          reshapeInComputationDim = idx;
+          break;
+        }
+        if (acc > computationRelevantSize) {
+          return rewriter.notifyMatchFailure(
+              matchOp, "bias/scale shape is not compatible with reshape input");
+        }
+      }
+
+      newScaleOrBiasShape.push_back(reshapeInShape[reshapeInComputationDim]);
+      newScaleOrBiasShape.append(
+          reshapeInShape.size() - reshapeInComputationDim - 1, 1);
+    }
+
+    // Norms only support unidirectional broadcasting to y
```

Comment:
```suggestion
    // Norms only support unidirectional broadcasting to x
```
Should it not be `x` ?

Reply:
Close review 👀 
I changed the comment